src/common/helpers/data-extract/data-extract.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @typedef  { 'name' | 'file' | 'text' |' radio' | 'address' | 'checkbox' } AnswerType
+ * @typedef  {{ type: AnswerType, value: string, displayText: string }} AnswerData
```

Comment:
Value isn't string in all supported cases - e.g. address, checkbox, or name. 

Might be better to do a type union of the answer types, so we can capture these differences: 

```javascript
/**
 * @typedef {{ type: 'name', value: { firstName: string, lastName: string }, displayText: string }} NameAnswer
 *  ....
 * @typedef { NameAnswer | AddressAnswer | ... } AnswerData
 */
```

Reply:
done